Fix Strict Date Parsing of ISO-8601 Dates Without Times

philjohnson789 opened the pull request:
When a valid ISO-8601 date string without a time component is parsed using `moment.ISO_8601` as the date format and strict parsing is enabled, the resulting Moment object is invalid.

Here's a JSFiddle: https://jsfiddle.net/k3snbte9/

mj1856 commented on src/test/moment/is_valid.js:
```diff
@@ -122,12 +122,14 @@ test('valid string iso 8601 + timezone', function (assert) {
         '2010-01-30T23:59:59.999+00:00',
         '2010-01-30T23:59:59.999-07:00',
         '2010-01-30T00:00:00.000+07:00',
-        '2010-01-30T00:00:00.000+07'
```
Technically, this is valid under ISO8601, but we don't currently support it.  See #1723.

philjohnson789 replied:
If we don't support it, why was there a unit test case? I had to remove it because it was failing when strict mode was enabled, so it seemed like a mistake in the tests.

ichernev replied:
Can we just add new test cases not modify existing ones? Now we're not testing 2 digit time zone.

ichernev commented:
Looks good just make sure not to drop existing tests.

Or fix them if they now fail :)

philjohnson789 commented:
As @mj1856 noted, that two-digit timezone test I removed was failing because the feature is not fully supported by the current version of Moment.js. When I added tests for strict mode, that test began to fail, as an unsupported feature should in strict mode.

Here's [the documentation from the MomentJS website](http://momentjs.com/docs/#/parsing/string-format/) that describes the functionality: `Offset from UTC as +-HH:mm, +-HHmm, or Z`.

For example, this is false: `moment('2015-06-30T09:36:00-05', moment.ISO_8601, true).isValid();`
Here's a JSFiddle that illustrates the behavior: https://jsfiddle.net/gwtd31qm/1/

I could add a separate test suite for inputs that are valid only when strict mode is false, but I think it makes more sense to just add full support for [the ISO 8601 offsets](https://en.wikipedia.org/?title=ISO_8601#Times) and then put the unit test back.

mj1856 commented:
Yes, they should be added back with the fix for #1723.

ichernev commented:
OK, I guess the 2 digit timezone will go in a separate PR.

ichernev commented:
Merged in ffecd46a175803f367e2363bf959ba6e7f2f45be
